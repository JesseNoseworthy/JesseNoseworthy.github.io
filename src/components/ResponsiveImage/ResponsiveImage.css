@import '../../styles/variables.css';
@import '../../styles/mixins.css';

:local(.ResponsiveImage) {
  
}