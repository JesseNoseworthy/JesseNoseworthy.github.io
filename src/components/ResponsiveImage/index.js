import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './ResponsiveImage.css';
import Image from 'components/Image';

class ResponsiveImage extends Component {
  getResponsiveImageSrc() {
    const {
      isDesktopAll,
      isTablet,
      isMobile,
      desktopImageSrc,
      tabletImageSrc,
      mobileImageSrc
    } = this.props;

    if (isDesktopAll) { return desktopImageSrc }
    else if (isTablet) { return tabletImageSrc || desktopImageSrc }
    else if (isMobile) { return mobileImageSrc || desktopImageSrc }
  }

  render() {
    const {
      className,
      alt,
      animationType,
      ariaHidden,
      ...others
    } = this.props;

    return (
      <Image
        className={classNames(styles.ResponsiveImage, className)}
        src={this.getResponsiveImageSrc()}
        ariaHidden={ariaHidden}
        alt={alt}
        animationType={animationType}
        {...others}
      />
    );
  }
}

ResponsiveImage.propTypes = {
  className: PropTypes.string,
  desktopImageSrc: PropTypes.string,
  tabletImageSrc: PropTypes.string,
  mobileImageSrc: PropTypes.string,
  alt: PropTypes.any.isRequired,
  isDesktopAll: PropTypes.bool,
  isTablet: PropTypes.bool,
  isMobile: PropTypes.bool,
  animationType: PropTypes.string,
  ariaHidden: PropTypes.string
};

export default ResponsiveImage;