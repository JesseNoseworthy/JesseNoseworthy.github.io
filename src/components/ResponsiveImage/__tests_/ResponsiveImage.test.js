import React from 'react';
import renderer from 'react-test-renderer';
import ResponsiveImage from '../index';

describe('ResponsiveImage', () => {
  describe('Snapshot tests', () => {
    it('default state (empty)', () => {
      const component = renderer.create(
        <ResponsiveImage 
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: false,
            isDesktopLarge: false,
            isDesktop: false,
            isTablet: false,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in desktop XL', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: true,
            isDesktopLarge: false,
            isDesktop: false,
            isTablet: false,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in desktop XL', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: true,
            isDesktopLarge: false,
            isDesktop: false,
            isTablet: false,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in desktop large', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: false,
            isDesktopLarge: true,
            isDesktop: false,
            isTablet: false,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in desktop standard', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: false,
            isDesktopLarge: false,
            isDesktop: true,
            isTablet: false,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in tablet', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: false,
            isDesktopLarge: false,
            isDesktop: false,
            isTablet: true,
            isMobile: false
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders in mobile', () => {
      const component = renderer.create(
        <ResponsiveImage
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
          responsiveFlags={{
            isDesktopExtraLarge: false,
            isDesktopLarge: false,
            isDesktop: false,
            isTablet: false,
            isMobile: true
          }}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
