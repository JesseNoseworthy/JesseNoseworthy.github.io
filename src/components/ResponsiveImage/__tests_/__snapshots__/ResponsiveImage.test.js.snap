exports[`ResponsiveImage Snapshot tests default state (empty) 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in desktop XL 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in desktop XL 2`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in desktop large 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in desktop standard 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in mobile 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;

exports[`ResponsiveImage Snapshot tests renders in tablet 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos="fade"
  src="my-img.jpg" />
`;
